### ISSUE TITLE
Advanced Cache Filter

### ISSUE THREAD

**Issue by aruehl** [cgeo/cgeo#6384]
I'm a fresh cacher and search the caches with my little children...

It is unnecessarily complicated and time-consuming if you search very specific caches.

My suggestion would be an advanced search, where the user can determine how many results are to be downloaded at the maximum (e.g. 1000), and agree that the search can take longer and the search results are stored on the device until the app closes.

On this data basis, a checkbox-based search is possible, in which you can select a wide range of criteria at the same time. For example:
type traditional and size large or regular and distance <= 10 km and difficulty 1 to 1.5

In the result list, I would like to be able to add several caches at one time to an offline list.

With the standard search, I've downloaded the same caches x times and then frustrated abandoned.

If you think that this is a minority wish: I have talked to five experienced cachers and they all said that they wanted it too some times.

**Comment by SammysHP** [comment]
This is a duplicate of #135.

First, we can only select if we want to download all cache types or a specific one. All other filters can only be applied after the caches were downloaded with all details (i.e. open or store them). This is a limitation of the interface that we use to connect to gc.com.

Another issue is the GUI which has a lot of usability issues.

And last: We can only download 20 caches at once. This is also a limitation of the interface with gc.com. c:geo updates data only if instructed so, it will not open additional result pages without user interaction. This is because it is intended as a special user interface for the website.

In summary: All mentioned issues are known and covered by other issues. Some of them cannot be fixed at all, others require lots of work and we would need someone who wants to implement it.

**Comment by aruehl** [comment]
This is not a duplicate of #135, because #135 wants only to combine two types, I want to combine several categories and store the results temporarily on the device. 

I now the download limit of 20 caches, but there is no limit of downloads.

**Comment by SammysHP** [comment]
> This is not a duplicate of #135

Then add #995, #5320, #5741, #5764 or #5877.

> I want to combine several categories and store the results temporarily on the device.

This won't be possible. All we could do is to download all caches and filter afterwards. Of course there's also #4977, but that would be another step after our internal search supports filter sets.

**Comment by Lineflyer** [comment]
@aruehl 
As @SammysHP linked there are some issues on the wishlist as well as some technical limitations.
If you would like to prepare your caching trips as you described a good possibility is http://www.project-gc.com where you can extendively search and filter and even create GPX files to import them to c:geo afterwards.
